src/main/java/com/google/sps/data/AvailabilityTimeSlots.java

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.TextStyle;
+import java.util.Locale;
+import java.time.format.DateTimeFormatter;
+
+// TODO: Add a test for the constructor.
+
+/**
+ * A collection of AvailabilityTimeSlot Objects, which will eventually be generated with information
+ * from datastore.
+ */
+public class AvailabilityTimeSlots {
+  private List<AvailabilityTimeSlot> timeSlots = new ArrayList<AvailabilityTimeSlot>();
+
+  /** Constructs an AvailabilityTimeSlots object by generating the timeSlots list. */
+  public AvailabilityTimeSlots(String timeZoneOffsetString) {
+    ZoneOffset timeZoneOffset = convertStringToOffset(timeZoneOffsetString);
+    ZoneId zoneId = ZoneId.ofOffset("UTC", timeZoneOffset);
+    ZonedDateTime today = Instant.now().atZone(zoneId);
+    String dayOfWeek = today.getDayOfWeek().getDisplayName(TextStyle.SHORT, Locale.US);
+    int year = today.getYear();
+    int month = today.getMonthValue();
+    int dayOfMonth = today.getDayOfMonth();
+
+    // TODO: Generate a week of slots.
+    String date = generateDate(dayOfWeek, month, dayOfMonth);
+    List<String> times = generateTimes();
+    List<String> utcEncodings = generateUTCEncodings(year, month, dayOfMonth, zoneId);
+    List<Boolean> selectedStatuses = getSelectedStatuses();
+    timeSlots = generateTimeSlots(utcEncodings, times, date, selectedStatuses);
+  }
+
+  private ZoneOffset convertStringToOffset(String offsetString) {
+    int offsetTotalMinutes = Integer.parseInt(offsetString);
+    int offsetHours = offsetTotalMinutes / 60;
+    int offsetMinutes = offsetTotalMinutes % 60;
+    return ZoneOffset.ofHoursMinutes(offsetHours, offsetMinutes);
+  }
+
+  private String generateDate(String dayOfWeek, int month, int dayOfMonth) {
+    return dayOfWeek + " " + month + "/" + dayOfMonth;
+  }
+
+  private List<String> generateUTCEncodings(int year, int month, int dayOfMonth, ZoneId zoneId) {
```

Comment:
I'd suggest you distinguish in the name between the functions here that generate fake data and the ones that generate real data.

Reply:
I'm still not entirely sure what you mean by fake data. Right now I have comments next to the getSelectedStatuses and generateDates functions because they are not fully implemented yet. What within this class qualifies as "real" and "fake"?

Reply:
Resolved with constants.